simplq/src/store/queues/queuesAsyncActions.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable no-param-reassign */
+import useAuth, { makeAuthedRequest } from 'api/auth';
+
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { QueueRequestFactory } from 'api/requestFactory';
+import * as RequestFactory from 'api/requestFactory';
+
+/**
+ * A hook to access the fetchQuees async action creator.
+ *
+ * @returns — fetchQueues async async action creator
+ *
+ * @example
+ *
+ * import useFetchQueues from 'store/queues'
+ *
+ * const fetchQueues = useFetchQueues()
+ */
+const useFetchQueues = () => {
+  const auth = useAuth();
+
+  const fetchQueues = createAsyncThunk('queues/requestStatus', async () => {
+    if (!auth || !auth.isAuthenticated) {
+      return { queues: [] };
+    }
+    const authedRequest = makeAuthedRequest(auth, QueueRequestFactory.getMyQueues());
+    const response = await authedRequest;
+    return response;
+  });
+
+  return fetchQueues;
+};
+
+/**
+ * A shadow function that returns fetchQueues async action creator.
+ *
+ * This should be used just for the action creators that it generates.
+
+ * @see ./queuesSlice
+ *
+ * @see https://redux-toolkit.js.org/api/createAsyncThunk#return-value
```

Comment:
I'll have to read this 😄

Reply:
@daltonfury42 where can I find deployed PR?

Reply:
I'm sorry, I didn't understand the last one.

Reply:
@mradenovic We generate temporary preview deployments. This can be triggered by putting any label on the PR (You can see a sample [here](https://github.com/SimplQ/simplQ-frontend/pull/441))

But due to some security limitations, this is not possible on PRs from forks.

Reply:
OK, I'll push to repo to see how it works.